### PR TITLE
Calculator Logic with tests

### PR DESCRIPTION
**Pull request by SamiSha99**
I am still waiting for a previous Pull request to be pulled so I can work on some few tests but this PR has the calculator while also connected to the data file for it.

~~Waiting for this PR to be merged #33~~

### CODE REVIEW

**Review by Amirk390 — APPROVED** [review]
looks good,
the test seems to work properly

File: src/lib/calculator.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// userChoice is how many activities have been clicked
+
+export default function calculator(data, userChoice) {
+  return Object.keys(userChoice).reduce(
+    (acc, cur) => {
+      data.activities.forEach((activity) => {
+        if (activity.name === cur) {
+          const liters = activity.liters * userChoice[cur];
+          acc.liters += liters;
```
Comment (matthewdking) [inline src/lib/calculator.js]: can make the two lines above into one!
```
acc.liters += activity.liters * userChoice[cur];
```

File: src/lib/calculator.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          const cost = activity.cost * userChoice[cur];
+          acc.cost += cost;
```
Comment (matthewdking) [inline src/lib/calculator.js]: Similar to above :point_up:

File: tests/test.js [hunk 3]
```diff
@@ -1,8 +1,53 @@
 import test from 'tape';
+import data from '../src/fixtures/data';
+import calculator from '../src/lib/calculator';
 
-test('1 = 1 ?', (t) => {
-  const actual = 1;
-  const expected = 1;
-  t.equal(actual, expected, 'Should return 1 when given 1');
+test('Test #0', (t) => {
```
Comment (matthewdking) [inline tests/test.js]: Change all test names! Descriptive naming is important.

File: tests/test.js [hunk 4]
```diff
@@ -1,8 +1,53 @@
...
+  const expected = {
+    liters: 9,
+    cost: 0.0801,
+  };
+
+  const actual = calculator(data, { toilet: 2 });
+  t.deepEqual(actual, expected);
```
Comment (matthewdking) [inline tests/test.js]: You should have a third argument here which is a string. This should be a descriptive error message that shows up when this test fails.

File: tests/test.js (hunk 3 above)
Comment (matthewdking) [reply tests/test.js]: This applies to all tests below :point_down:

File: tests/test.js (hunk 4 above)
Comment (matthewdking) [reply tests/test.js]: This applies to all tests below :point_down:

File: tests/test.js [hunk 5]
```diff
@@ -1,8 +1,53 @@
...
+    liters: 107.5,
+    cost: 0.95675,
+  };
+
+  const actual = calculator(data, {
+    toilet: 5,
+    shower: 7,
+    'Faucet tap': 3,
+    'Washing machine': 2,
+    Dishwasher: 1,
```
Comment (matthewdking) [inline tests/test.js]: I think this is partly my fault for missing it in a previous PR. But I don't like the inconsistency in this object. Starting with a mixture of upper and lower case, also I personally dislike keys with multiple words with spaces

**Review by matthewdking — CHANGES_REQUESTED** [review]

File: tests/test.js [hunk 6]
```diff
@@ -1,8 +1,64 @@
...
+    Shower: 7,
+    'Faucet tap': 3,
+    'Washing machine': 2,
+    Dishwasher: 1,
+  });
+  t.deepEqual(actual, expected, 'Test Passed, results are 107 liters and a price of 0.95675');
+  t.end();
+});
+
+test('Shiny looking calculate button I wonder what it will do?', (t) => {
```
Comment (matthewdking) [inline tests/test.js:55]: :laughing:

**Review by matthewdking — APPROVED** [review]
Nice :ok_hand:

**Pull request merged**